packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -286,14 +293,9 @@ function WrappedDay<TDate extends PickerValidDate>({
       'data-timestamp': utils.toJsDate(day).valueOf(),
       ...other,
     },
-    ownerState: { ...parentProps, day, selected: isSelected },
+    ownerState: { ...ownerState, isDayDisabled: isDisabled, isDaySelected: isSelected },
```

Comment:
That one is up for discussion.
I removed `day` because I don't like the idea of passing objects like this in the `ownerState`, but if some people need to style based on the object (and not just on some properties that we can extract), then I can add it back of course.

I also think this object should eventually be the one that is passed to the `styleOverrides` that @noraleonte is re-working (so that all customization API around day have the same `ownerState`).
So I'm only adding things that for me will obviously be in the one @noraleonte is working on.